src/viewport.cpp

```diff
@@ -1214,62 +1218,81 @@ void ViewportAddString(const DrawPixelInfo *dpi, ZoomLevel small_from, const Vie
 	}
 }
 
-static void ViewportAddTownNames(DrawPixelInfo *dpi)
+static void ViewportAddKdtreeSigns(DrawPixelInfo *dpi)
 {
-	if (!HasBit(_display_opt, DO_SHOW_TOWN_NAMES) || _game_mode == GM_MENU) return;
+	/* These values are heuristic, it's better to catch too many items than too few */
+	int left   = _vd.dpi.left - 500;
+	int top    = _vd.dpi.top - 100;
```

Comment:
I'd appreciate some help in finding a better heuristic, or plain safer data source, for what range to search in.

Reply:
For the height: `ScaleByZoom(VPSM_TOP + FONT_HEIGHT_NORMAL + VPSM_BOTTOM, dpi->zoom)`
For the width: I think you have to store the max width for all the signs, and update this max on all update virt coords calls.